packages/waku/src/lib/builder/serve-partykit.ts

```diff
@@ -0,0 +1,19 @@
+import { Hono } from 'hono';
+
+import { honoMiddleware } from '../middleware/hono-prd.js';
+
+const ssr = !!import.meta.env.WAKU_BUILD_SSR;
+const loadEntries = () => import(import.meta.env.WAKU_ENTRIES_FILE!);
+let serveWaku: ReturnType<typeof honoMiddleware> | undefined;
+
+const app = new Hono();
+
+app.use('*', (c, next) => serveWaku!(c, next));
+export default {
+  onFetch(request: Request, lobby: any, ctx: Parameters<typeof app.fetch>[2]) {
+    if (!serveWaku) {
+      serveWaku = honoMiddleware({ loadEntries, ssr, env: lobby });
```

Comment:
btw, this was the only workaround I came up with for cloudflare.
How do you think this is feasible in the long run?

Reply:
This is fine. Another option would be to have a helper function that generates the export. Yet another option would be to use AsyncLocalStorage. I'll send a PR later if you'd like to clean this up.

Reply:
If such a solution works for both --with-cloudflare and --with-partykit, I would be interested. I don't think AsyncLocalStorage would work, because it's a Node.js API.

Reply:
AsyncLocalStorage is now available in cloudflare with their node compat mode (which is turned on by default in partykit) https://developers.cloudflare.com/workers/runtime-apis/nodejs/

Reply:
currently, we don't use node compat mode for cloudflare deploy.

what's your second option? > a helper function that generates the export.